localstack/services/apigateway/apigateway_listener.py

```diff
@@ -152,6 +152,15 @@ def authorize_invocation(api_id, headers):
         run_authorizer(api_id, headers, authorizer)
 
 
+def is_api_key_valid(is_api_key_required, headers):
+    if not is_api_key_required:
+        return True
+
```

Comment:
Note that the header name should be `X-API-Key`: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-key-source.html

Also, can we leave a TODO here:
```
# TODO: Currently, we're only checking the presence of the API key in the header.
# In the future we need to verify the actual key value against the REST API configuration as well.
```